File: app/src/main/java/com/suzei/minote/ui/list/ListContract.java
package com.suzei.minote.ui.list;


import com.suzei.minote.data.entity.Notes;

import java.util.List;

public interface ListContract {

    interface View {

        void setPresenter(Presenter presenter);

        void showListOfNotes(List<Notes> listOfNotes);

        void showListUnavailable();

        void insertNoteToList(Notes note, int position);

        void redirectToEditorActivity(int itemId);

    }

    interface Presenter {

        void start();

        void deleteNote(Notes note);


        void showNoteEditor(int itemId);

    }

}

File: app/src/main/java/com/suzei/minote/ui/list/ListContract.kt
package com.suzei.minote.ui.list


interface ListContract {

    interface View<T> {

        fun setPresenter(presenter: Presenter<T>)

        fun showListOfNotes(listOfNotes: MutableList<T>)

        fun showListUnavailable()

    }

    interface Presenter<T> {

        fun setup()

        fun delete(data: T)

    }

}

interface ToastCallback {

    fun onUndoClick()
    fun onToastDismiss()

}